docs/writing/posts/lgtmk.md

```diff
@@ -0,0 +1,229 @@
+
+**Example via RAG**
+
+- **Slow Metric**: Collocating human preferences and consulting domain experts.
+- **Still Slow Metric**: AI-generated metrics. When using something like GPT4, things can become very slow.
+- **Fast Metrics**: Accuracy, Precision, Recall, MRR, NDCG, are computationally cheap given the labels.
+
+The goal is to reason about the trade-offs between fast metrics and slow data. It takes a long time to get enough data so you can move fast. But if you never do that work, we're always gonna be stuck.
+
+## Simple Metrics for Relevancy and Ranking
+
+In the retrieval context, there are plenty of metrics to choose from. I'm gonna go describe a couple of them. But before we do that, we need to understand what @k means.
+
+### Understanding @K
+
+The simplest idea we should think about is the idea of @k. When we do RAG, we first have to retrieve a set of K documents. Then we will do some re-ranking potentially. And then select the top end results to show to a user or to a language model. Consider the following pipeline:
+
+1. Fetch n documents via Keyword Search
+2. Fetch n documents via Semantic Search
+3. Combine them and re-rank
+4. Select the top 25 chunks to show to LLM
+5. Top 5 documents are shown to the user.
+
+```mermaid
+graph LR
+    X[Query] --> A[Keyword Search];
+    X --> B[Semantic Search];
+    B --> C[Reranker]
+    A --> C[Reranker]
+    C --> D[Top 25]
+    C --> E[Top 5]
+    D --> Y[LLM]
+    E --> Z[User]
+```
+
+Now let's look at some interpretations of top-k results.
+
+| k   | Interpretation                      |
+| --- | ----------------------------------- |
+| 5   | Is what we show the user relevant?  |
+| 25  | Is the reranker doing a good job?   |
+| 50  | Is the retrieval system doing well? |
+| 100 | Did we have a shot at all?          |
+
+I strongly recommend you not focus too hard on generation from an LLM, And to stay focused on being able to provide the right context. You will be able to get a language model to be more robust and as language models improve, they will only get more resilient to irrelevant information. However, as you build out your business, this data set that you curate on relevancy will stay with you.
+
+## Thinking about the metrics @ K
+
+Now let's look at some metrics that we can use to evaluate the performance of our retrieval augmented generation system. The goal isn't to give a mathematical breakdown of these metrics, but instead give you a sense of what they mean and how they might be useful. And how I like to explain and interpret them at the limits.
+
+### Mean Average Recall (MAR) @ K
+
+Focuses on the system's capability to retrieve all relevant documents within the top K results, emphasizing the breadth of relevant information captured.
```

Comment:
probably should note that the total number of relevant documents is a known unknown, as the search is usually incomplete/time-bound in a large enough system